lib/filewatch/watched_file.rb

```diff
@@ -0,0 +1,180 @@
+require "filewatch/buftok"
+
+module FileWatch
+  class WatchedFile
```

Comment:
If a record exists in Watch `@files` then it is being "watched".  The Ruby File may not be attached or open.  I think of this as the internal record of a path and inode, holding the state (and history) no matter how many times the actual file was opened and closed.  Only when the path cannot Stat will the record be destroyed.  When someone adds a pattern to the `@exclude` array then that makes a watched_file become excluded (unwatched) and its arguable whether we should delete it from the `@files` hash at that time because I don't think one can undo exclusion.  Also when someone calls `unwatch(path)` then that makes an watched_file become unwatched and again one ca't undo an `unwatch`.  BTW, `unwatch` was added to support a read use case. It can probably be removed.